app/src/main/java/org/mozilla/fenix/components/toolbar/ToolbarUIView.kt

```diff
@@ -87,6 +87,7 @@ class ToolbarUIView(
 
     override fun updateView() = Consumer<SearchState> {
         if (it.isEditing) {
+            view.url = it.query
```

Comment:
We only want to set the url based on the query if in editMode. This is because when transitioning to displayMode it's possible the session is not yet set (and thus has no query). For example, when making a new tab and navigating to the first page, the updateView is called before a new sessionID is generated to grab a URL from, which results in the URL being set to "".